core/src/upickle/core/Types.scala

```diff
@@ -277,16 +277,16 @@ trait Types{ types =>
         ctx.visitEnd(-1)
       }
     }
-    protected def writeSnippet[R, V](objectAttributeKeyWriteMap: CharSequence => CharSequence,
-                                     ctx: _root_.upickle.core.ObjVisitor[_, R],
-                                     mappedArgsI: String,
-                                     w: Writer[V],
-                                     value: V) = {
+    def writeSnippet[R, V](objectAttributeKeyWriteMap: CharSequence => CharSequence,
```

Comment:
I couldn't figure out how to subclass the path dependent `prefix.CaseW` type in the macro, so I had to instantiate it in the inline part first and then call into the macro after. That means the call to `writeSnippet` is no longer from the subclass, and it thus can no longer be protected